grammars/json.gbnf

```diff
@@ -16,10 +16,10 @@ array  ::=
 string ::=
   "\"" (
     [^"\\\x7F\x00-\x1F] |
-    "\\" (["\\/bfnrt] | "u" [0-9a-fA-F] [0-9a-fA-F] [0-9a-fA-F] [0-9a-fA-F]) # escapes
+    "\\" (["\\bfnrt] | "u" [0-9a-fA-F]{4}) # escapes
   )* "\"" ws
 
-number ::= ("-"? ([0-9] | [1-9] [0-9]*)) ("." [0-9]+)? ([eE] [-+]? [0-9]+)? ws
+number ::= ("-"? ([0-9] | [1-9] [0-9]{0,15})) ("." [0-9]+)? ([eE] [-+]? [0-9] [1-9]{0,15})? ws
 
 # Optional space: by convention, applied in this grammar after literal chars when allowed
-ws ::= ([ \t\n] ws)?
+ws ::= [ \t\n]{,20}
```

Comment:
This feels like a good change -- I like constraining the output in this way. Could even consider limiting it to something more restrictive like `{,4}` or `{,8}`, but this is a good start.

Reply:
I've drafted an updated space rule in https://github.com/ggerganov/llama.cpp/pull/7866. No matter what the bound is with this current syntax, model seem keen to misuse it. But given near-unlimited indent space only (and only 1 newline at a time), they're very sensible.